### DISCUSSION TITLE
Add shader complexity preview in viewport and visual profiler

### DISCUSSION THREAD

**Discussion by Foyezes** [godotengine/godot-proposals#11374] [Editor]
### Describe the project you are working on

3d game that uses shaders

### Describe the problem or limitation you are having in your project

Currently there's no way of knowing how complex a shader is (that I know of)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

A viewport view mode that displays shader complexity similar to UE. A visual profiler frame time will be useful as well.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

See above

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No

### Is there a reason why this should be core and not an add-on in the asset library?

This feature should be core

**Comment by Calinou** [comment]
Moving to discussion, as there is no concrete technical implementation provided.
